staging/src/k8s.io/apiserver/pkg/admission/audit.go

```diff
@@ -28,6 +29,10 @@ import (
 type auditHandler struct {
 	Interface
 	ae *auditinternal.Event
+	// Indicate whether concurrency protection is required
+	concurrencyProtection bool
```

Comment:
i think it's not very costly to always have concurrency on - so i would recommend removing the bool

Reply:
I have checked all the call points of the `audit.LogAnnotation`. Currently, only calls from `auditHandler`  trigger this problem. However, I must admit that it is safer to move the lock inside `audit.LogAnnotation`. I'll revise it.

Reply:
> but I haven't seen sync.Mutex declared in types.go before, should I declare the lock as a global variable?

you are right, the types do not have any internal states.  moving the lock inside `audit.LogAnnotation` is not trivial so let's address it in a separate PR.

I think what you have right now should resolve the panic we are seeing. We also need to back port this PR to  supported releases, so let's keep it short so it is back portable.

Reply:
> should I declare the lock as a global variable?

that would not be at request scoped, so all requests in flight would fight for this lock which is not good :)